### PR TITLE
Speed up CI by splitting crates into batches

### PR DESCRIPTION
**Pull request by jdisanti**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/ci/get-ci-matrix.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import json
+import math
+import os
+import shlex
+import subprocess
+import sys
+import unittest
+
+DESIRED_WORKERS = 6
```
Comment (rcoh) [inline tools/ci/get-ci-matrix.py:19]: might want to make this configurable via env-var

File: tools/ci/get-ci-matrix.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+def main():
+    if len(sys.argv) == 1:
+        print(f"Usage: {sys.argv[0]} <rust versions to use...>")
+        return 1
+
+    rust_versions = sys.argv[1:]
+    repository_root = get_cmd_output("git rev-parse --show-toplevel")
+    os.chdir(repository_root)
+
+    crate_paths = get_cmd_output("./tools/ci/list-sdk-crate-paths.sh").splitlines(keepends=False)
```
Comment (rcoh) [inline tools/ci/get-ci-matrix.py:31]: would probably keep this all in-script unless you have a compelling reason for it to be separate (`os.listdir`, etc.)

File: tools/ci/get-ci-matrix.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
```
Comment (rcoh) [inline tools/ci/get-ci-matrix.py:1]: you should be able to make a `util.py` to share utilities between the different scripts

**Review by rcoh — APPROVED** [review]
LGTM—I'd probably refactor the Python scripts a bit so that they just call functions instead of invoking each other and reparsing

File: tools/ci/get-ci-matrix.py (hunk 2 above)
Comment (jdisanti) [reply tools/ci/get-ci-matrix.py:31]: Yeah, the separate scripts are an artifact of my incremental approach to solving the problem. I should clean it up!

File: .github/workflows/audit.yaml [hunk 4]
```diff
@@ -1,11 +1,11 @@
 name: Security audit
 on:
   schedule:
-    - cron: '0 0 * * *'
+    - cron: "0 0 * * *"
   workflow_dispatch:
   pull_request:
 env:
-  rust_version: 1.51.0
+  rust_version: 1.54.0
```
Comment (Velfi) [inline .github/workflows/audit.yaml:8]: Can we add a note to the release guide once we ship this? I don't want us to forget to update this every 2 weeks

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -42,14 +42,14 @@ jobs:
...
+      - name: Remove root manifest
         run: |
           # HACK: The workspace smithy-rs generates is invalid.
           # This can be removed once https://github.com/awslabs/smithy-rs/pull/957
           # is merged into aws-sdk-rust.
           rm Cargo.toml
-
-          ./tools/ci/crate-range.py run ${{ matrix.crate_range }} cargo test
+      - name: Cargo Test
+        run: python3 -u ./tools/ci/crate-range.py run ${{ matrix.crate_range }} cargo test
```
Comment (rcoh) [inline .github/workflows/ci.yaml:52]: you can add the `-u` in the shebang to save future folks headache

**Review by rcoh — APPROVED** [review]

**Pull request merged**